haskell-src/Concordium/Types/Transactions.hs

```diff
@@ -139,6 +141,60 @@ instance S.Serialize TransactionSignature where
                     accumulateSigs (Map.insert idx sigmap accum) (Just idx) (count - 1)
         TransactionSignature <$> accumulateSigs Map.empty Nothing len
 
+-----------------------------------------------------------------
+
+-- * JSON representation of a signed/partially-signed transaction
+
+-----------------------------------------------------------------
+
+-- | A 'SignedTransaction' is a transaction that is signed by an account (the signer)
+-- with some keys. The representation might be a fully signed transaction ready to be
+-- sent on-chain or a partially-signed transaction that needs additional signatures
+-- added to be ready to be sent on-chain.
+--
+-- The `ToJSON` instance has the purpose converting the object into a human-readable
+-- representation ready to be printed into a JSON file. This file can be shared among
+-- different tools of the Concordium ecosystem for adding additional signatures.
+--
+--  The chosen representation is the minimal necessary data needed to construct the
+-- 'TransactionSignHash' which is the value that is signed by the signer. The
+-- 'TransactionSignHash' and 'payloadSize' should be re-computed when processing a
+-- 'SignedTransaction' (e.g. when adding signatures or sending the transaction on-chain).
+--
+-- The representation has a `version` field.
+data SignedTransaction = SignedTransaction
```

Comment:
I am not sure why this should be here (in base) rather than in the client.

Reply:
The main reason to move it to base was that this type should be used across Concordium components (meaning the multi-sig JSON format should be compatible). E.g. maybe we add a service to the `wallet-proxy` for wallets to display pending partially-signed transactions at some point and this type can be re-used.

Reply:
`concordium-client` is a dependency of the `wallet-proxy` anyway, so this is not a very strong reason. (`concordium-client` is sort of the Haskell SDK + CLI.)

Reply:
`Concordium-client` should be a CLI and not be treated as a `Haskell SDK` for other components to depend on it.

Reply:
ok, that makes sense, if we argue that `base` should be kept minimal and only include a necessary dependency for the node especially since the `base` is at the end of some of our sub-modules link chains.
That `concordium-client` is treated as a Haskell SDK is something we should fix at some point so :).